src/ripple/rpc/handlers/ServerInfo.cpp

```diff
@@ -22,13 +22,313 @@
+        {
+            std::string type_name =
+                translate(str(pair.first).substr(4) /* remove STI_ */);
+            int32_t type_value = std::stoi(str(pair.second));
+            ret[jss::TYPES][type_name] = type_value;
+            type_map[type_value] = type_name;
+        }
+
+        ret[jss::LEDGER_ENTRY_TYPES] = Json::objectValue;
+        ret[jss::LEDGER_ENTRY_TYPES][jss::Invalid] = -1;
+
+        for (const auto& pair : LedgerFormats::getInstance().getNamesAndTypes())
+            ret[jss::LEDGER_ENTRY_TYPES][str(pair.first)] =
+                std::stoi(str(pair.second));
+
+        ret[jss::FIELDS] = Json::arrayValue;
+
+        uint32_t i = 0;
+        {
+            Json::Value a = Json::arrayValue;
+            a[0U] = "Generic";
+            Json::Value v = Json::objectValue;
+            v[jss::nth] = 0;
+            v[jss::isVLEncoded] = false;
+            v[jss::isSerialized] = false;
+            v[jss::isSigningField] = false;
+            v[jss::type] = "Unknown";
+            a[1U] = v;
+            ret[jss::FIELDS][i++] = a;
+        }
+
+        {
+            Json::Value a = Json::arrayValue;
+            a[0U] = "Invalid";
+            Json::Value v = Json::objectValue;
+            v[jss::nth] = -1;
+            v[jss::isVLEncoded] = false;
+            v[jss::isSerialized] = false;
+            v[jss::isSigningField] = false;
+            v[jss::type] = "Unknown";
+            a[1U] = v;
+            ret[jss::FIELDS][i++] = a;
+        }
+
+        {
+            Json::Value a = Json::arrayValue;
+            a[0U] = "ObjectEndMarker";
+            Json::Value v = Json::objectValue;
+            v[jss::nth] = 1;
+            v[jss::isVLEncoded] = false;
+            v[jss::isSerialized] = true;
+            v[jss::isSigningField] = true;
+            v[jss::type] = "STObject";
+            a[1U] = v;
+            ret[jss::FIELDS][i++] = a;
+        }
+
+        {
+            Json::Value a = Json::arrayValue;
+            a[0U] = "ArrayEndMarker";
+            Json::Value v = Json::objectValue;
+            v[jss::nth] = 1;
+            v[jss::isVLEncoded] = false;
+            v[jss::isSerialized] = true;
+            v[jss::isSigningField] = true;
+            v[jss::type] = "STArray";
+            a[1U] = v;
+            ret[jss::FIELDS][i++] = a;
+        }
+
+        {
+            Json::Value a = Json::arrayValue;
+            a[0U] = "taker_gets_funded";
+            Json::Value v = Json::objectValue;
+            v[jss::nth] = 258;
+            v[jss::isVLEncoded] = false;
+            v[jss::isSerialized] = false;
+            v[jss::isSigningField] = false;
+            v[jss::type] = "Amount";
+            a[1U] = v;
+            ret[jss::FIELDS][i++] = a;
+        }
+
+        {
+            Json::Value a = Json::arrayValue;
+            a[0U] = "taker_pays_funded";
+            Json::Value v = Json::objectValue;
+            v[jss::nth] = 259;
+            v[jss::isVLEncoded] = false;
```

Comment:
what is the rationale for setting these fields to `true/false`? How are you parsing the source code?

Reply:
I don't understand your question. 
Here's an example of the output of this RPC in a library: https://github.com/XRPLF/xrpl.js/blob/main/packages/ripple-binary-codec/src/enums/definitions.json
All this data is needed so that libraries know how to encode and decode transactions/ledger objects/other binary data that's in the custom rippled serialization format. 
The scripts that are currently used to generate these files are here: https://github.com/RichardAH/xrpl-codec-gen

Reply:
okay thanks